source/MaterialXRender/External/glTF_MaterialX/GltfMaterialHandler.h

```diff
@@ -0,0 +1,225 @@
+    {
+        return _extensions;
+    }
+
+    /// <summary>
+    ///     Translate shaders in the document to appropriate shading model(s)
+    ///     for export. Derived classes can override this method
+    ///     to perform actions such as shader translation and baking.
+    /// </summary>
+    /// <param name="doc">Document to modify</param>
+    /// <param name="log">Error log</param>
+    virtual void translateShaders(DocumentPtr doc, StringVec& log) 
+    {
+        std::ignore = doc;
+        std::ignore = log;
+    };
+
+    /// <summary>
+    ///     Set document containing MaterialX definitions. This includes core library
+    ///     definitions
+    /// </summary>
+    /// <param name="doc">Definition document</param>
+    void setDefinitions(DocumentPtr doc)
+    {
+        _definitions = doc;
+    }
+
+    /// <summary>
+    ///     Set document to use for MaterialX material generation or extraction.
+    /// </summary>
+    /// <param name="materials">MaterialX document</param>
+    void setMaterials(DocumentPtr materials)
+    {
+        _materials = materials;
+    }
+
+    /// <summary>
+    ///     Get MaterialX document containing material information
+    /// </summary>
+    /// <returns>MaterialX document</returns>
+    DocumentPtr getMaterials() const
+    {
+        return _materials;
+    }
+
+    /// <summary>
+    ///     Set whether to generate MaterialX assignments if found in the input glTF file.
+    ///     By default assignments are not generated.
+    /// </summary>
+    /// <param name="val">Generate assignments flag</param>
+    void setGenerateAssignments(bool val)
+    {
+        _generateAssignments = val;
+    }
+
+    /// <summary>
+    ///     Get whether to generate MaterialX material assignments.
+    /// </summary>
+    /// <returns>True if generating assignments</returns>
+    bool getGenerateAssignments() const
+    {
+        return _generateAssignments;
+    }
+
+    /// <summary>
+    ///     Set whether to generate all inputs on MaterialX nodes when converting from glTF file.
+    ///     By default all inputs are generated.
+    /// </summary>
+    /// <param name="val">Generate inputsflag</param>
+    void setGenerateFullDefinitions(bool val)
+    {
+        _generateFullDefinitions = val;
+    }
+
+    /// <summary>
+    ///     Get whether to generate all inputs for MaterialX nodes.
+    /// </summary>
+    /// <returns>True if generating</returns>
+    bool getGenerateFullDefinitions() const
+    {
+        return _generateFullDefinitions;
+    }
+
+  protected:
+    StringSet _extensions;
+    DocumentPtr _definitions;
+    DocumentPtr _materials;
+
+    // Generation options
+    bool _generateFullDefinitions;
+    bool _generateAssignments;
+    bool _generateNodeDefs;
+};
+
+/// @class GltfMaterialHandler
+/// Wrapper for handling import / export of materials to / from GLTF files
+class MX_GLTF_API GltfMaterialHandler : public MaterialHandler
```

Comment:
If JSON export is added and used for conversion then it can derive from the `MaterialHandler` base class as needed.